test/integration/placement_test.go

```diff
@@ -591,6 +628,44 @@ var _ = ginkgo.Describe("Placement", func() {
 
 		})
 
+		ginkgo.It("Should schedule successfully based on AddOnPlacementScore", func() {
```

Comment:
I think we also need cases including

1. an addon score is missing and added
2. addon score is updated

You could set resync interval to a variable so you can shorten the resync period here.

Also it worth to run the benchmark testing with addon score enabled but that could be a followup.

Reply:
Add a reschedule case for AddOnPlacementScore changes. And will log an issue to trace the benchmark testing.